tensorbay/cli/ls.py

```diff
@@ -50,19 +50,25 @@ def _validate_dataset_client(dataset_client: DatasetClientType) -> None:
         error(f'"{dataset_client.name}" has no commit history.')
 
 
-def _ls_dataset(gas: GAS, info: TBRN, list_all_files: bool) -> None:
+def _ls_dataset(gas: GAS, info: TBRN, list_all_files: bool, show_total_num: bool) -> None:
     dataset_client = get_dataset_client(gas, info)
     _validate_dataset_client(dataset_client)
 
     segment_names = dataset_client.list_segment_names()
     if not list_all_files:
+        if show_total_num:
+            click.echo(f"total {segment_names.__len__()}")
         for segment_name in segment_names:
             click.echo(TBRN(info.dataset_name, segment_name).get_tbrn())
         return
 
     if isinstance(dataset_client, FusionDatasetClient):
         error('"-a" flag is not supported for fusion dataset yet')
-
+    if show_total_num:
+        total_num = 0
+        for segment_name in segment_names:
+            total_num += dataset_client.get_segment(segment_name).list_data_paths().__len__()
```

Comment:
Should we store the results of `dataset_client.get_segment(segment_name).list_data_paths()` to avoid duplicate requests in `echo_data`?